### PR TITLE
[I18n] RBAC refactoring form interactions

### PR DESCRIPTION
**Pull request by soupette**
### What does it do?

It simplifies the logic of the `<Permissions />` component's reducer for the collection types and the single types tabs.

### CODE REVIEW

File: packages/strapi-admin/ee/admin/containers/Roles/CreatePage/Permissions/reducer.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+       *   }
+       *  }
+       * }
+       *
+       * The value of the create checkbox for the restaurant will be ƒalse since not all its children have
+       * truthy values and in order to set its value to true when need to have all the values of its children set to true.
+       *
+       * Similarly, we can reuse the logic for the components attributes
+       *
+       */
```
Comment (HichamELBSI) [inline packages/strapi-admin/ee/admin/containers/Roles/CreatePage/Permissions/reducer.js:107]: 👍🏽

File: packages/strapi-admin/ee/admin/containers/Roles/CreatePage/Permissions/utils/createDefaultCTFormFromLayout.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  }, {});
+};
+
+export default createDefaultCTFormFromLayout;
+export {
+  createDefaultConditionsForm,
+  createDefaultPropertyForms,
+  createDefaultPropertiesForm,
+  findLayouts,
+};
```
Comment (HichamELBSI) [inline packages/strapi-admin/ee/admin/containers/Roles/CreatePage/Permissions/utils/createDefaultCTFormFromLayout.js:136]: very nice, can you just rename the `acc` and `acc2` variables to something more understandable.

File: packages/strapi-admin/ee/admin/containers/Roles/CreatePage/temp/fakeData.js [hunk 3]
```diff
@@ -398,19 +404,20 @@ const data = {
...
           actionId: 'content-manager.explorer.update',
-          subjects: ['addresse', 'restaurant'],
+          subjects: ['address', 'erestaurant'],
           applyToProperties: ['fields'],
         },
         {
           label: 'Delete',
           actionId: 'content-manager.explorer.delete',
           subjects: ['restaurant', 'address'],
+          applyToProperties: [],
```
Comment (HichamELBSI) [inline packages/strapi-admin/ee/admin/containers/Roles/CreatePage/temp/fakeData.js:420]: Do you still need this file?

**Review by HichamELBSI — CHANGES_REQUESTED** [review]
Good job @soupette

File: packages/strapi-admin/ee/admin/containers/Roles/CreatePage/temp/fakeData.js (hunk 3 above)
Comment (soupette) [reply packages/strapi-admin/ee/admin/containers/Roles/CreatePage/temp/fakeData.js:420]: Yes! the backend is not ready yet.

**Pull request merged**